compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -501,10 +505,31 @@ private void copyJar(ZipArchiveOutputStream outStream, JarLibrary jarLibrary,
 
         // Transfers selected entries from this zip file to the output stream, while preserving its compression and
         // all the other original attributes.
-        zipFile.copyRawEntries(outStream, predicate);
+        zipFile.copyOptimisedRawEntries(outStream, predicate);
+//        zipFile.copyRawEntries(outStream, predicate);
         zipFile.close();
     }
 
+    private static HashMap<String, HashSet<String>> getDeadFunctionList(String pkgName, CompilerContext compilerContext) {
+        org.wso2.ballerinalang.compiler.PackageCache packageCache = org.wso2.ballerinalang.compiler.PackageCache.getInstance(compilerContext);
```

Comment:
Import the `PackageCache` and use it here

Reply:
Will change it

Reply:
A different class with the same name (`PackageCache`) is already imported. So we will have to go with this for now.